src/main/java/gregtech/client/ClientProxy.java

```diff
@@ -189,12 +188,19 @@ public static void addMaterialFormulaHandler(ItemTooltipEvent event) {
 
         // Test for Items
         UnificationEntry unificationEntry = OreDictUnifier.getUnificationEntry(itemStack);
-        if (unificationEntry != null && unificationEntry.material != null) {
-            chemicalFormula = unificationEntry.material.getChemicalFormula();
-
-        // Test for Fluids
-        } else if (ItemNBTUtils.hasTag(itemStack)) {
 
+        if (itemStack.getItem() instanceof MetaOreDictItem) { // Test for OreDictItems
+            MetaOreDictItem oreDictItem = (MetaOreDictItem) itemStack.getItem();
+            Optional<String> oreDictName = OreDictUnifier.getOreDictionaryNames(itemStack).stream().findFirst();
+            if (oreDictName.isPresent() && oreDictItem.OREDICT_TO_FORMULA.containsKey(oreDictName.get())) {
+                String foundFormula = oreDictItem.OREDICT_TO_FORMULA.get(oreDictName.get());
+                if (foundFormula != null) {
```

Comment:
This would not be a null formula, but rather `"""` according to your comment in `MetaOreDictItem#calculateChemicalFormula`